colearn_grpc/utils.py

```diff
@@ -116,11 +116,20 @@ def transform_training_summary_to_pb(ts: TrainingSummary) -> TrainingSummaryPb:
 
 def transform_training_summary_from_pb(ts: TrainingSummaryPb) -> TrainingSummary:
     tsn = TrainingSummary()
-    for field in ["dp_budget"]:
-        target = getattr(tsn, field)
+    parts = [
+        ("dp_budget", DiffPrivBudget, {
+            "target_epsilon": 0,
+            "target_delta": 0,
+            "consumed_epsilon": 0,
+            "consumed_delta": 0
+        })
+    ]
+    for (field, cls, default_params) in parts:
```

Comment:
Maybe its just me, but I find it a bit difficult to read this function's code.

Also, the `error_code` should not be parsed here too?
https://github.com/fetchai/colearn/blob/4d7c80fcb5053dcc523ee3ffd625b07716304ffe/colearn/ml_interface.py#L58

Reply:
not on this level. this simply do decodes.

it is a more complex structure, so later on can support other fields in TrainingSummary